### DISCUSSION TITLE
contributing/development/handling_compatibility_breakages

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#123] [User-contributed notes]
# contributing/development/handling_compatibility_breakages

So you've added a new parameter to a method, changed the return type, changed the type of a parameter, or changed its default value, and now the automated testing is complaining about compatibility...

https://docs.godotengine.org/en/latest/contributing/development/handling_compatibility_breakages.html

**Comment by aaronfranke** [comment]
Note: It's not possible to do this for core Variant types yet: https://github.com/godotengine/godot-proposals/issues/10337

**Reply by surendra019** [reply]
could you please tell me where is the extension_api.json file for the output?

from the top of this file-

https://github.com/godotengine/godot/blob/master/misc/extension_api_validation/4.2-stable_4.3-stable.expected

**Reply by Calinou** [reply]
This file is generated using the `--dump-extension-api` [command line argument](https://docs.godotengine.org/en/stable/tutorials/editor/command_line_tutorial.html) on a Godot editor binary. The Godot version matters here, as the generated extension API file is different for each Godot version.

You can find a pregenerated file that loosely follows Godot's `master` branch here: https://github.com/godotengine/godot-cpp/blob/master/gdextension/extension_api.json

You can check stable branches for files that will work with stable releases like 4.3.x: https://github.com/godotengine/godot-cpp/blob/4.3/gdextension/extension_api.json
